test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertFrom-Json.Tests.ps1

```diff
@@ -1,6 +1,32 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
-Describe 'ConvertFrom-Json' -tags "CI" {
+
+function New-NestedJson {
+    Param([int] $Depth)
+
+    $nestedJson = "true"
+
+    $Depth..1 | ForEach-Object {
+        $nestedJson = '{"' + $_ + '":' + $nestedJson + '}'
+    }
+
+    return $nestedJson
+}
+
+function Count-ObjectDepth {
+    Param([PSCustomObject] $InputObject)
+
+    for ($i=1; $i -le 100000; $i++)
```

Comment:
Since you have a max depth for checking here, you should probably enforce this limit on `new-nestedjson`.  Also, I'm not sure we need to test this deep, perhaps 2048 is sufficient